bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/rest/Schedules.java

```diff
@@ -0,0 +1,342 @@
+        ruleRegistry.addRegistryChangeListener(this);
+
+        for (Rule item : ruleRegistry.getAll()) {
+            added(item);
+        }
+    }
+
+    @Deactivate
+    public void deactivate() {
+        ruleRegistry.removeRegistryChangeListener(this);
+    }
+
+    /**
+     * Called by the registry when a rule got added (and when a rule got modified).
+     * <p>
+     * Converts the rule into a {@link HueScheduleEntry} object and add that to the hue datastore.
+     */
+    @Override
+    public void added(Rule rule) {
+        if (!rule.getTags().contains(SCHEDULE_TAG)) {
+            return;
+        }
+        HueScheduleEntry entry = new HueScheduleEntry();
+        entry.name = rule.getName();
+        entry.description = rule.getDescription();
+        entry.autodelete = rule.getActions().stream().anyMatch(p -> p.getId().equals("autodelete"));
+        entry.status = ruleManager.isEnabled(rule.getUID()) ? "enabled" : "disabled";
+
+        String timeStringFromTrigger = RuleUtils.timeStringFromTrigger(rule.getTriggers());
+        if (timeStringFromTrigger == null) {
+            logger.warn("Schedule from rule '{}' invalid!", rule.getName());
+            return;
+        }
+
+        entry.localtime = timeStringFromTrigger;
+
+        for (Action a : rule.getActions()) {
+            if (!a.getTypeUID().equals("rules.HttpAction")) {
+                continue;
+            }
+            HueCommand command = RuleUtils.httpActionToHueCommand(cs.ds, a, rule.getName());
+            if (command == null) {
+                continue;
+            }
+            entry.command = command;
+        }
+
+        cs.ds.schedules.put(rule.getUID(), entry);
+    }
+
+    @Override
+    public void removed(Rule element) {
+        cs.ds.schedules.remove(element.getUID());
+    }
+
+    @Override
+    public void updated(Rule oldElement, Rule element) {
+        removed(oldElement);
+        added(element);
+    }
+
+    /**
+     * Creates a new rule that executes a http rule action, triggered by the scheduled time
+     *
+     * @param uid A rule unique id.
+     * @param builder A rule builder that will be used for creating the rule. It must have been created with the given
+     *            uid.
+     * @param oldActions Old actions. Useful if `data` is only partially set and old actions should be preserved
+     * @param data The configuration for the http action and trigger time is in here
+     * @return A new rule with the given uid
+     * @throws IllegalStateException If a required parameter is not set or if a light / group that is referred to is not
+     *             existing
+     */
+    protected static Rule createRule(String uid, RuleBuilder builder, List<Action> oldActions,
+            List<Trigger> oldTriggers, HueChangeScheduleEntry data, HueDataStore ds) throws IllegalStateException {
+        HueCommand command = data.command;
+        Boolean autodelete = data.autodelete;
+
+        String temp;
```

Comment:
I'm not such `temp` as a name :-)

Reply:
I can't do anything here unfortunately except renaming the variable to something else which doesn't fix the problem. It is enforced by the nullness library that we are using to create a local variable copy.